pkg/console/controllers/resourcesyncdestination/controller.go

```diff
@@ -27,118 +26,103 @@ const (
 )
 
 type ResourceSyncDestinationController struct {
+	operatorClient v1helpers.OperatorClient
+	// operatorconfig
 	operatorConfigClient operatorclientv1.ConsoleInterface
-	configMapClient      coreclientv1.ConfigMapsGetter
 	// events
-	cachesToSync []cache.InformerSynced
-	queue        workqueue.RateLimitingInterface
-	recorder     events.Recorder
-	// context
-	ctx context.Context
+	resourceSyncer resourcesynccontroller.ResourceSyncer
 }
 
 func NewResourceSyncDestinationController(
+	operatorClient v1helpers.OperatorClient,
 	// operatorconfig
 	operatorConfigClient operatorclientv1.ConsoleInterface,
 	operatorConfigInformer operatorinformersv1.ConsoleInformer,
-	// configmap
-	corev1Client coreclientv1.CoreV1Interface,
-	configMapInformer coreinformersv1.ConfigMapInformer,
+	// openshift managed
+	managedCoreV1 coreinformerv1.Interface,
 	// events
 	recorder events.Recorder,
-	// context
-	ctx context.Context,
-) *ResourceSyncDestinationController {
-	corev1Client.ConfigMaps(api.OpenShiftConsoleNamespace)
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+) factory.Controller {
+	managedConfigMapInformer := managedCoreV1.ConfigMaps()
 
 	ctrl := &ResourceSyncDestinationController{
+		operatorClient:       operatorClient,
 		operatorConfigClient: operatorConfigClient,
-		configMapClient:      corev1Client,
 		// events
-		recorder:     recorder,
-		cachesToSync: nil,
-		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
-		ctx:          ctx,
+		resourceSyncer: resourceSyncer,
 	}
 
-	configMapInformer.Informer().AddEventHandler(ctrl.newEventHandler())
-	operatorConfigInformer.Informer().AddEventHandler(ctrl.newEventHandler())
-	ctrl.cachesToSync = append(ctrl.cachesToSync,
-		operatorConfigInformer.Informer().HasSynced,
-		configMapInformer.Informer().HasSynced,
-	)
-
-	return ctrl
+	configNameFilter := util.NamesFilter(api.ConfigResourceName)
+
+	return factory.New().
+		WithFilteredEventsInformers( // configs
+			configNameFilter,
+			operatorConfigInformer.Informer(),
+		).WithFilteredEventsInformers(
+		util.NamesFilter(api.DefaultIngressCertConfigMapName),
```

Comment:
this does not seem to be the only CM that gets synced in this controller

Reply:
there is another custom-logo CM, but it's name is set by the admin on the console-operator config, and we read it from there. Since the name is arbitrary we can't set the informer.

Reply:
in that case, you need to watch all in the given NS, see `WithNamespaceInformer()`